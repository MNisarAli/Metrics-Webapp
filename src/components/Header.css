.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 0.6rem;
  background-color: #d6447d;
  border-color: #fff3;
  font-size: 1.3rem;
}

.nav-left,
.nav-right {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.nav-icon {
  font-size: 1.5rem;
  cursor: pointer;
}

.year,
.title {
  font-size: 1.3rem;
  font-weight: 700;
}

.icon-mic {
  font-size: 1.2rem;
}
