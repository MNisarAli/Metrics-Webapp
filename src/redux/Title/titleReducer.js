const UPDATE_TITLE = 'UPDATE_TITLE';

export const updateTitle = (newTitle) => ({
  type: UPDATE_TITLE,
  title: newTitle,
});

const initialState = {
  title: 'Initial Title',
};

export default function titleReducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_TITLE:
      return {
        ...state,
        title: action.title,
      };
    default:
      return state;
  }
}
