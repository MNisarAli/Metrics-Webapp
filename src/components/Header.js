import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import './Header.css';
import { updateTitle } from '../redux/Title/titleReducer';

const Header = () => {
  // Initialize variables and hooks.
  const dispatch = useDispatch();
  const location = useLocation();
  const { title } = useSelector((state) => state.title);
  const year = new Date().getFullYear();

  // Update the title of the header based on the current page.
  useEffect(() => {
    let newHeading = '';

    switch (location.pathname) {
      case '/':
        newHeading = 'World Metrics';
        break;
      case '/continent':
        newHeading = 'Continent Details';
        break;
      case '/country':
        newHeading = 'Country Details';
        break;
      default:
        newHeading = 'Page Not Found';
        break;
    }

    dispatch(updateTitle(newHeading));
  }, [location, dispatch]);

  return (
    <header className="header" data-testid="header">
      <div className="nav-left">
        <h2 className="year" data-testid="year">{year}</h2>
      </div>
      <h1 className="title">{title}</h1>
      <div className="nav-right">
      </div>
    </header>
  );
};

export default Header;
