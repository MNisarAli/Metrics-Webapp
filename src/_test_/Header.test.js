import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/configureStore';
import '@testing-library/jest-dom';
import Header from '../components/Header';
import { updateTitle } from '../Redux/Title/titleReducer';

describe('updateTitle', () => {
  it('should return the expected action object', () => {
    const newTitle = 'World Population';
    const expectedAction = {
      type: 'UPDATE_TITLE',
      title: newTitle,
    };

    expect(updateTitle(newTitle)).toEqual(expectedAction);
  });
});

describe('Header component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
