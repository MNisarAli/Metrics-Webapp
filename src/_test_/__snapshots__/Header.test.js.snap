// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header
  className="header"
  data-testid="header"
>
  <div
    className="nav-left"
  >
    <svg
      className="nav-icon"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
      />
    </svg>
    <h2
      className="year"
      data-testid="year"
    >
      2023
    </h2>
  </div>
  <h1
    className="title"
  >
    Initial Title
  </h1>
  <div
    className="nav-right"
  >
    <svg
      className="nav-icon icon-mic"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
      />
      <path
        d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
      />
    </svg>
    <svg
      className="nav-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
      />
    </svg>
  </div>
</header>
`;
